src/gui/dialogs/statistics_dialog.cpp

```diff
@@ -154,34 +156,35 @@ void statistics_dialog::add_damage_row(
 
 	const long long shifted = ((expected * 20) + shift) / (2 * shift);
 	std::ostringstream str;
-	str << damage << " / "
-		<< static_cast<double>(shifted) * 0.1
-		<< spacer // TODO: should probably make this two columns
-		<< (((dsa < 0) ^ (expected < 0)) ? "" : "+")
-		<< (expected == 0 ? 0 : 100 * dsa / expected) << '%';
-
+	str << damage << " / " << static_cast<double>(shifted) * 0.1;
 	item["label"] = str.str();
 	data.emplace("damage_overall", item);
 
 	str.str("");
+	str << (((dsa < 0) ^ (expected < 0)) ? "" : "+")
+		<< (expected == 0 ? 0 : 100 * dsa / expected) << '%';
+	item["label"] = str.str() + spacer;
+	data.emplace("overall_percent", item);
 
 	if(show_this_turn) {
 		const long long turn_shifted = ((turn_expected * 20) + shift) / (2 * shift);
-		str << turn_damage << " / "
-			<< static_cast<double>(turn_shifted) * 0.1
-			<< spacer // TODO: should probably make this two columns
-			<< (((dst < 0) ^ (turn_expected < 0)) ? "" : "+")
-			<< (turn_expected == 0 ? 0 : 100 * dst / turn_expected) << '%';
-
+		str.str("");
+		str << turn_damage << " / " << static_cast<double>(turn_shifted) * 0.1;
 		item["label"] = str.str();
 		data.emplace("damage_this_turn", item);
+
+		str.str("");
+		str << (((dst < 0) ^ (turn_expected < 0)) ? "" : "+")
+			<< (turn_expected == 0 ? 0 : 100 * dst / turn_expected) << '%';
+		item["label"] = str.str() + spacer;
+		data.emplace("this_turn_percent", item);
 	}
 
 	damage_list.add_row(data);
 }
 
-// Return the string to use in the "Hits" table, showing actual and expected number of hits.
-static std::string tally(const std::map<int, struct statistics::stats::by_cth_t>& by_cth)
+// Return the strings to use in the "Hits" table, showing actual and expected number of hits.
+static std::pair<std::string, std::string> tally(const std::map<int, struct statistics::stats::by_cth_t>& by_cth)
```

Comment:
Please create a named struct instead of a pair.

Reply:
Created in c6e7bb5e0bb.